draft-trammell-quic-spin.md

```diff
@@ -710,6 +710,41 @@ are actually within the detectable range.
   (both) the signal being detected and govern the non-uniform sampling
   frequency.
 
+# Greasing
+
+Routes, congestion levels and therefore latency between two fixed QUIC
+endpoints, as well as the shape of individual application flows, fluctuate in
+ways that are not totally predictable by an on path observer.  In general,
+there is no a-priori pattern for the spin-bit distribution that will always
+materialise on a certain flow aggregate, even for a single user.
+
+There has been discussion in the QUIC working group that greasing could be a
+strategy to counter an evil access provider that might gate access to its users
+on a valid spin bit signal.  Let's accept for a moment this threat model and
+consider the practical case of a home gateway that temporarily misbehaves, for
+example draining its queues slower than it'd normally do while a firmware
+download is in progress.  It would be ill-considered for an access provider
+(even a malicious one) to block, or otherwise interfere with, QUIC flows
+originating from behind that CPE solely based on the fact that RTTs are now
+different from "usual".  And that simply because of the sheer number of false
+negatives that the operator would get by setting an arbitrary threshold.
+
+So, the potential for ossification appears to be extremely low - in a sense,
+since it depends on so much external noise, the spin-bit is self-greasing.  In
+fact, implementing explicit greasing around the spin-bit might even be harmful
+as it'd potentially erode confidence in the veracity of the signal.
+
+However, if a greasing algorithm is really needed - for example, if we'd want
+to reuse the bit with different semantics in the future (i.e.: the spin-bit is
+not included in the header invariants), one very simple implementation would be
+as follows: each server will refuse to spin its bit on a per-flow basis with a
+given probability p, instead leaving it stuck to a randomly chosen value, 0
+or 1.  The client will then end up leaving its bit stuck to the opposite, or
+could detect this condition and also pick a randomly chosen stuck value.  The
```

Comment:
"could detect this condition and also pick a randomly chosen stuck value."<<< I'm wondering whether there is actual freedom to choose a value at random, and I think that a server which does not follow the convention of initially sending "0" will actually cause spin-bit flips instead of constant values if an arbitrary client follows the #mechanism.

Reply:
Since the client and server both initialize to 0, the client *should* be able to detect "stuck at 1" if the first spin it receives from the server is 1, no?

Reply:
We mention here that greasing could help ensure the bit can be used for other purposes in the future. Would this mechanism really be effective for that? What we get here is ossification around spin-bit enabled when the bit value is not static throughout a connection, or spin-bit disabled when the values are static. 
If the future use of the bit is not static, then we haven´t gained anything by doing the greasing in the first place? Or am I misunderstanding what we are protecting against when we grease the field like this?

Reply:
I tend to agree.  My take here is that the bit is self-greasing and doesn't need anything extra.

Reply:
@britram :  I agree, although client receiving first packet with spin=1, is unexpected from a compliant server. A continued stream of packets with spin=1 would then indicate stuck at 1.
And, I see a flaw in my logic from last night: a normal Client Receiving packet with spin=1 should flip it's next packet to spin=0, but it's already sending packets with spin=0, and in the initial condition the Client has sent No packets with spin=1.
If we keep this form of greasing, I think we need to describe the "error/stuck" condition in #mechanism.